challenge3/webapp/caesaris/app/lib/data_process.rb

```diff
@@ -0,0 +1,77 @@
+class DataProcess
+  class << self
+    # calc average energy_production per house for cities
+    # res: { "Oxford" => { "2011-7" => [min, ave, max] } }
+    def house_energy_prod_time_series
+      results = {}
+      City.all.each do |city|
+        res_ttl = {}
```

Comment:
```suggestion
        res_ttl = Hash.new do |h, key|
          h[key] = []
        end
```
こうしておくと、12行目の nil だった場合に `[]` を代入する処理を
不要にできる。

Reply:
そもそも、 res_ttl というハッシュを用意する必要はなく、二要素の配列を使えば十分なのではないだろうか。